actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/actor-util/src/main/java/org/sunbird/actorutil/email/impl/EmailServiceClientImpl.java

```diff
@@ -23,18 +26,27 @@ public Response sendMail(ActorRef actorRef, Map<String, Object> requestMap) {
     request.put(JsonKey.EMAIL_REQUEST, requestMap);
     actorRequest.setOperation((String) requestMap.get(JsonKey.REQUEST));
     actorRequest.setRequest(request);
+    Response response = null;
+    try {
+      Timeout t = new Timeout(Duration.create(10, TimeUnit.SECONDS));
+      Future<Object> future = Patterns.ask(actorRef, request, t);
+      Object obj = Await.result(future, t.duration());
 
-    Object obj = interServiceCommunication.getResponse(actorRef, actorRequest);
-    if (obj instanceof Response) {
-      Response response = (Response) obj;
-      return response;
-    } else if (obj instanceof ProjectCommonException) {
-      throw (ProjectCommonException) obj;
-    } else {
-      throw new ProjectCommonException(
-          ResponseCode.SERVER_ERROR.getErrorCode(),
-          ResponseCode.SERVER_ERROR.getErrorMessage(),
-          ResponseCode.SERVER_ERROR.getResponseCode());
+      if (obj instanceof Response) {
+        response = (Response) obj;
+      } else if (obj instanceof ProjectCommonException) {
+        throw (ProjectCommonException) obj;
```

Comment:
No need to else if and else condition if we are catching all exception and throwing differnet exception like unableToCommunicateWithActor exception.

Reply:
Here we are checking the response obj type and throwing the exception. Added Projectcommon exception throw clause in try catch